R/zchunk_L2231.land_input_3_irr.R

```diff
@@ -0,0 +1,356 @@
+      min.veg.carbon.density <- min.soil.carbon.density <- . <- NULL
+
+
+    # Process inputs
+    # Replace GLU names and Add region names as needed
+    L121.CarbonContent_kgm2_R_LT_GLU %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      replace_GLU(map = basin_to_country_mapping) %>%
+      rename(mature.age = `mature age`) ->
+      L121.CarbonContent_kgm2_R_LT_GLU
+
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L125.LC_bm2_R_LT_Yh_GLU
+
+
+    # Build tables
+
+    # L223.LN3_Logit: Logit exponent of the third nest.
+    # First, Determine the node combinations applicable at this level.
+    # Then, match in logit exponents based on the land node 2.
+    # Finally, append GLU names and keep only relevant columns
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      select(region, GLU, Land_Type) %>%
+      # not all land types have node matches, so use left_join
+      left_join(select(A_LT_Mapping, Land_Type, LandNode1, LandNode2, LandNode3),
+                               by = "Land_Type") %>%
+      select(-Land_Type) %>%
+      distinct() %>%
+      na.omit() %>%
+      mutate(LandAllocatorRoot = "root",
+             logit.year.fillout = min(BASE_YEARS)) %>%
+      # logit.type is NA by default, so left_join
+      left_join(select(A_LandNode_logit, logit.exponent, logit.type, LandNode), by = c("LandNode3" = "LandNode")) %>%
+      append_GLU(var1 = "LandNode1", var2 = "LandNode2", var3 = "LandNode3") %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["LN3_Logit"]], "logit.type"))) ->
+      L223.LN3_Logit
+
+
+    # Land Use History
+
+    # Unmanaged land tables
+    #
+    # These tables are formed from a master table, made by filtering and adding
+    # node_leaf_names in L125.LC.
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      filter(Land_Type %in% A_LandLeaf_Unmgd3$UnmanagedLandLeaf,
+             year %in% c(LAND_HISTORY_YEARS, BASE_YEARS)) %>%
+      mutate(allocation = round(value, aglu.DIGITS_LAND_USE)) %>%
+      add_node_leaf_names(nesting_table = A_LandLeaf_Unmgd3, leaf_name = "UnmanagedLandLeaf",
+                          LN1 = "LandNode1", LN2 = "LandNode2", LN3 = "LandNode3") ->
+      L223.LC_bm2_R_Unmgd3_Yh_GLU
+
+    # L223.LN3_HistUnmgdAllocation: Historical land cover, unmanaged land in the third nest
+    # L223.LN3_UnmgdAllocation: Model base period land cover, unmanaged land in the third nest
+    # Formed from filtering the master table by different years.
+    L223.LC_bm2_R_Unmgd3_Yh_GLU %>%
+      filter(year %in% LAND_HISTORY_YEARS) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_HistUnmgdAllocation"]])) ->
+      L223.LN3_HistUnmgdAllocation
+
+    L223.LC_bm2_R_Unmgd3_Yh_GLU %>%
+      filter(year %in% BASE_YEARS) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["LN3_UnmgdAllocation"]])) ->
+      L223.LN3_UnmgdAllocation
+
+
+    # Managed land - non-crop (forest)
```

Comment:
Currently, this is just `non-crop (forest)`, but really this code would do anything we make a `LandLeaf` at level 3. Maybe change the comment to reflect?